saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SemanticRoleLabeling/SRLApps.scala

```diff
@@ -47,25 +52,30 @@ object SRLApps extends Logging {
 
   val modelJars = properties.getString(SRLConfigurator.SRL_JAR_MODEL_PATH)
 
-  val expName: String = {
-    if (trainingMode.equals("other"))
-      if (trainArgType && useGoldBoundaries && useGoldPredicate && trainingMode.equals("other")) "aTr"
+  val experimentName: String = {
+    if (trainingMode.equals("other")) {
+      if (trainArgType && useGoldBoundaries && useGoldPredicate) "aTr"
       else if (trainArgIdentifier && useGoldPredicate && useGoldPredicate) "bTr"
       else if (trainArgType && useGoldPredicate && !useGoldBoundaries) "cTr"
       else if (trainPredicates && useGoldPredicate) "dTr"
       else if (trainArgIdentifier && !useGoldPredicate) "eTr"
       else if (trainArgType && !useGoldPredicate) "fTr"
       else ""
```

Comment:
Might be unnecessary because your `require(experimentName.nonEmpty, "Experiment is not configured properly.")` down below....